components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -1444,4 +1459,28 @@ private void validateAttributeMappings(List<AttributeMappingDTO> attributeMappin
             }
         }
     }
+
+    private void validateClaimModificationEligibility() throws ClaimMetadataClientException {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            String organizationId = getOrganizationManager().resolveOrganizationId(tenantDomain);
+            String rootOrganizationId = getOrganizationManager().getPrimaryOrganizationId(organizationId);
```

Comment:
Instead of getting the primary organization out shall we call "isPrimaryOrganization(orgid)" method ?
It makes the logic simple

Reply:
fixed